include/albatross/src/core/dataset.hpp

```diff
@@ -95,6 +95,32 @@ deduplicate(const RegressionDataset<FeatureType> &dataset) {
   return albatross::subset(dataset, unique_inds);
 }
 
+template <typename X>
+inline auto align_datasets(RegressionDataset<X> *x, RegressionDataset<X> *y) {
```

Comment:
Name of the function is a bit strange to me. "Align" on its own doesn't mean much to me, and the loss of data is important here. "Intersect" would be good.
`aligned_dataset_intersection`? `intersect_datasets`?

Reply:
I was mirroring the vocab used in pandas: https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.DataFrame.align.html

I think unless you're very opposed to `align` I'd like to stick to it.  I'd be OK with `align_datasets_on_intersection` though.